metric/metrics.go

```diff
@@ -24,25 +25,41 @@ const (
 )
 
 // Set is the basic structure for storing metrics.
-type Set map[string]interface{}
+type Set struct {
+	// The storer is required for deltas and rates, but forces us overriding marshaller and makes unmarshaller
+	// impossible. We should set back metrics. Set to a simple map, and manage it from a metrics.SetManager class, or the
+	// Integration class.
+	// This forces us, for the WHOLE integration. Do a clear separation between logic and data.
+	// TODO: global refactor, separating for all the classes, the JSON payload from the entities that manage it
+	storer  persist.Storer
+	Metrics map[string]interface{}
+}
 
 // NewSet returns a new Set instance.
-func NewSet(eventType string) Set {
-	ms := Set{}
+// TODO: this function should not be visible to the SDK user (move it to another package?)
```

Comment:
I guess this TODO is temporary. Maybe work for another PR? Could you give me a brief summary of how the user will add metric sets? Just to be aligned.

Reply:
Yes, my intention was to remove it in another PR.

Reply:
And yes, My idea was to add metrics through a proxy function.